### DISCUSSION TITLE
Stream backed by callbacks

### DISCUSSION THREAD

**Discussion by stephenaa** [nmoinvaz/minizip#779] [Q&A]
For a network application I would like to have a stream that is backed by callbacks, for this specefic case really only writes are interesting. Is there any examples of such a use of the library or will I have to create my own stream implementation?

I imagine something like a way to set a size of a memory buffer, and a callback to call once that is filled with data - having the callback return how much data was consumed and decrease the amount of data in the buffer accordingly.

**Comment by nmoinvaz** [comment]
The `vtbl` for `mz_stream_s` is essentially just a bunch of callbacks.
